cmd/contour/certgen.go

```diff
@@ -137,34 +138,47 @@ func OutputCerts(config *certgenConfig, kubeclient *kubernetes.Clientset, certs
 		case "compact":
 			secrets = certgen.AsSecrets(config.Namespace, certs)
 		default:
-			check(fmt.Errorf("unsupported Secrets format %q", config.Format))
+			return fmt.Errorf("unsupported Secrets format %q", config.Format)
 		}
 	}
 
 	if config.OutputPEM {
 		fmt.Printf("Writing certificates to PEM files in %s/\n", config.OutputDir)
-		check(certgen.WriteCertsPEM(config.OutputDir, certs, force))
+		if err := certgen.WriteCertsPEM(config.OutputDir, certs, force); err != nil {
+			return fmt.Errorf("Writing certificates to PEM files in %s/\n", config.OutputDir)
+		}
 	}
 
 	if config.OutputYAML {
 		fmt.Printf("Writing %q format Secrets to YAML files in %s/\n", config.Format, config.OutputDir)
-		check(certgen.WriteSecretsYAML(config.OutputDir, secrets, force))
+		if err := certgen.WriteSecretsYAML(config.OutputDir, secrets, force); err != nil {
+			return fmt.Errorf("failer to write %q format Secrets to YAML files in %s/\n", config.Format, config.OutputDir)
+		}
 	}
 
 	if config.OutputKube {
 		fmt.Printf("Writing %q format Secrets to namespace %q\n", config.Format, config.Namespace)
-		check(certgen.WriteSecretsKube(kubeclient, secrets, force))
+		if err := certgen.WriteSecretsKube(kubeclient, secrets, force); err != nil {
+			return fmt.Errorf("failed to write  %q format Secrets to namespace %q\n", config.Format, config.Namespace)
+		}
 	}
+	return nil
 }
 
-func doCertgen(config *certgenConfig) {
+func doCertgen(config *certgenConfig, log logrus.FieldLogger) {
+
 	generatedCerts, err := GenerateCerts(config)
-	check(err)
+	if err != nil {
+		log.WithError(err).Fatal("failed to generate certs")
+	}
 
 	clients, err := k8s.NewClients(config.KubeConfig, config.InCluster)
 	if err != nil {
-		check(fmt.Errorf("failed to create Kubernetes client: %w", err))
+		log.WithError(err).Fatalf("failed to create Kubernetes client: %v", err)
+	}
+
+	if oerr := OutputCerts(config, clients.ClientSet(), generatedCerts, log); oerr != nil {
```

Comment:
Rename `oerr` to `err`. Note that this causes a bug because the log message below uses `err` :)

Reply:
Replaced the error with oerr.